### ISSUE TITLE
Xerces 3.2 error in conda install

### ISSUE THREAD

**Issue by bittremieux** [OpenMS/OpenMS#4369]
When I install OpenMS via conda by default the latest 3.2.2 version of Xerces is installed as a dependency. However, it seems that this version is not compatible (possibly related to #3073?), giving the following error:

```
$ FeatureFinderCentroided --help
FeatureFinderCentroided: error while loading shared libraries: libxerces-c-3.1.so: cannot open shared object file: No such file or directory
```

When I downgrade Xerces to version 3.1.4 it seems to work.

**Comment by timosachsenberg** [comment]
Thanks for reporting. I also had the impression this had been fixed.
I can definitely say that installation using conda worked before.
For now, I think we could pin the xerces-c version in the bioconda recipe to 3.1.4:
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/openms/meta.yaml
@Leon-Bichmann can you take a look if that works and update the recipe?

**Comment by Leon-Bichmann** [comment]
you mean like this:

https://github.com/bioconda/bioconda-recipes/pull/18776

**Comment by bittremieux** [comment]
Maybe just do version `3.1` without the patch number to allow any subversion of 3.1?

Also, take care that it's version 3.1, not 1.3. 😉

**Comment by Leon-Bichmann** [comment]
haha seems like I had a number twister there ^^

**Comment by jpfeuffer** [comment]
The 2.4 release only supports automatic finding of boost 1.65 or lower since they changed their default naming layout.
https://github.com/OpenMS/OpenMS/commit/4da97faeb0c2874c37f3c3024fab473fe0a19a70

**Comment by Leon-Bichmann** [comment]
bioconda fails

**Comment by jpfeuffer** [comment]
see the comment above ;)

**Comment by Leon-Bichmann** [comment]
so do you mean I would have to set:
 boost =< 1.65 ?

**Comment by jpfeuffer** [comment]
yes I would try that

**Comment by Leon-Bichmann** [comment]
still sth. wrong?

**Comment by jpfeuffer** [comment]
I guess it wasn't the finding of boost but the different c++ standard it was compiled with.

**Comment by Leon-Bichmann** [comment]
ok I guess we got to solve this in another way..

**Comment by oliveralka** [comment]
@jpfeuffer Any idea how to fix the boost linking error:

Boost version 1.65.0

https://circleci.com/gh/bioconda/bioconda-recipes/83549?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link

```
.13:04:36 BIOCONDA INFO (OUT) [ 80%] Linking CXX executable ../../bin/MapAlignerSpectrum
13:04:36 BIOCONDA INFO (OUT) /opt/conda/conda-bld/openms_1574425533954/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::match_results<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >::maybe_assign(boost::match_results<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > const&)'
13:04:36 BIOCONDA INFO (OUT) /opt/conda/conda-bld/openms_1574425533954/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::re_detail_106500::perl_matcher<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::construct_init(boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags)'
13:04:36 BIOCONDA INFO (OUT) /opt/conda/conda-bld/openms_1574425533954/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::re_detail_106500::cpp_regex_traits_implementation<char>::transform[abi:cxx11](char const*, char const*) const'
13:04:36 BIOCONDA INFO (OUT) /opt/conda/conda-bld/openms_1574425533954/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::re_detail_106500::cpp_regex_traits_implementation<char>::transform_primary[abi:cxx11](char const*, char const*) const'
13:04:36 BIOCONDA INFO (OUT) collect2: error: ld returned 1 exit status
13:04:36 BIOCONDA INFO (OUT) make[3]: *** [bin/MapAlignerSpectrum] Error 1
```

**Comment by oliveralka** [comment]
Ok, now I get it (with the explanation of Timo). The issue here overlaps with the issue that all boost versions in conda are compiled with c++14, that why we get the reference linking error with std::__cxx11. 
So it seems that the "hotfix" using  `-DCMAKE_CXX_STANDARD=14` to force it to use c++14 everywhere. Do you have any other ideas to fix this?

**Comment by oliveralka** [comment]
Julianus had the suggestion to use the release branch instead of the tar.gz in terms of the source code for bioconda. 

For example:
```
{% set version = "2.4.0" %}
source:
    git_url: https://github.com/OpenMS/OpenMS
    git_tag: release/{{version}}
```

This would give us the possibility to backport #4379, to support (a) the used compile version or (b) manual setting of the compiler cxx standard, which would hopefully solve the boost issue.

Steps:
1)  backport
2) use release/{{version}}
If needed set  "-DCMAKE_CXX_STANDARD"  to 14

**Comment by jpfeuffer** [comment]
The fix that was used in conda for pyOpenMS 2.4 was to replace the CXX Standard in our CMakeLists with a patch.
I have a fixed recipe for openms that I will push soon. For newer releases we only set a minimal version via compile features.
